### PR TITLE
Article list

### PR DESCRIPTION
**Pull request by brunnermanu**

### CODE REVIEW

File: src/app/articles/article-detail/article.component.ts [hunk 1]
```diff
@@ -0,0 +1,10 @@
+import { Component } from '@angular/core';
+
+@Component({
+  selector: 'app-article',
+  templateUrl: './article.component.html',
+  styleUrls: ['./article.component.scss']
+})
+export class ArticleComponent {
+
+}
```
Comment (khanlifts) [inline src/app/articles/article-detail/article.component.ts]:
```suggestion
export class ArticleComponent {}
```

File: src/app/articles/articles-list/article-pre-view/article-pre-view.component.html [hunk 2]
```diff
@@ -0,0 +1,15 @@
+<div class="articlePreView">
+  <img class="articlePreView__title-image" src="{{article.titleImage}}" alt="title-image">
+  <div class="articlePreView__title">{{article.title}}</div>
+  <div class="author">
+    <img class="author__profile-picture" src="{{article.authorImage}}" alt="profile-picture">
+
```
Comment (khanlifts) [inline src/app/articles/articles-list/article-pre-view/article-pre-view.component.html]: Nice markup!
1. Do you want to add spaces in the interpolations ike `{{ abc }}`?
2. Remove this empty line?

File: src/app/articles/articles-list/article-pre-view/article-pre-view.component.scss [hunk 3]
```diff
@@ -0,0 +1,57 @@
+@import "src/styles/variables";
+@import "src/styles/maps";
+@import "src/styles/mixins";
+
+.articlePreView {
+	width: 250px;
+	height: 350px;
+	background-color: $white;
+	border-radius: 10px;
```
Comment (khanlifts) [inline src/app/articles/articles-list/article-pre-view/article-pre-view.component.scss]: Do you want to repurpose the `border-radius` and move it into a map?

File: src/app/articles/articles-list/article-pre-view/article-pre-view.component.ts [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+  selector: 'app-article-pre-view',
+  templateUrl: './article-pre-view.component.html',
+  styleUrls: ['./article-pre-view.component.scss']
+})
+
+
+export class ArticlePreViewComponent {
+  @Input() article: Article;
+  @Input() index: number;
+
```
Comment (khanlifts) [inline src/app/articles/articles-list/article-pre-view/article-pre-view.component.ts]: empty line

File: src/app/articles/articles-list/article-pre-view/article-pre-view.component.ts [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+@Component({
+  selector: 'app-article-pre-view',
+  templateUrl: './article-pre-view.component.html',
+  styleUrls: ['./article-pre-view.component.scss']
+})
+
+
+export class ArticlePreViewComponent {
+  @Input() article: Article;
+  @Input() index: number;
```
Comment (khanlifts) [inline src/app/articles/articles-list/article-pre-view/article-pre-view.component.ts]: Can you help me see the purpose of this `index` input?

File: src/app/articles/articles-list/articles.component.scss [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+@import "src/styles/maps";
+@import "src/styles/mixins";
+
+.articles {
+	padding-top: $layout-spacing-large;
+	padding-bottom: $layout-spacing-large;
+	@include page-styles(standard-spaces);
+
+	&__title {
+		@include headings(h2);
```
Comment (khanlifts) [inline src/app/articles/articles-list/articles.component.scss]: Do you know why you have `headings(h2)` in a `h4` tag?

File: src/app/articles/articles-list/articles.component.ts [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+    } else {
+      this.articles = this.articlesService.getArticles()
+    }
+  }
+
+  onArticleClick(itemId: number): void {
+    this.router.navigate(['/articles', itemId]).then(a => a);
+  }
+
+
```
Comment (khanlifts) [inline src/app/articles/articles-list/articles.component.ts]: empty lines

File: src/app/articles/articles-list/articles.component.ts [hunk 8]
```diff
@@ -0,0 +1,32 @@
...
+  ngOnInit() {
+    if (this.showLatest) {
+      this.articles = this.articlesService.getArticles().slice(-3);
+    } else {
+      this.articles = this.articlesService.getArticles()
+    }
+  }
+
+  onArticleClick(itemId: number): void {
+    this.router.navigate(['/articles', itemId]).then(a => a);
```
Comment (khanlifts) [inline src/app/articles/articles-list/articles.component.ts]:
```suggestion
    void this.router.navigate(['/articles', itemId]);
```

File: src/app/home/home.component.ts [hunk 9]
```diff
@@ -1,10 +1,25 @@
...
+  constructor(private articlesService: ArticlesService, private router: Router) {
+  }
+
+  ngOnInit() {
+    this.articles = this.articlesService.getArticles()
+  }
+
+  onArticleClick(itemId: number): void {
+    this.router.navigate(['/articles', itemId]).then(r => r);
+  }
```
Comment (khanlifts) [inline src/app/home/home.component.ts]: You can probably delete all of this logic.

File: src/app/services/articles.service.ts [hunk 10]
```diff
@@ -0,0 +1,52 @@
...
+      createdAt: "2.1.2023",
+      authorName: "Manuela",
+      authorImage: "assets/images/manu-profile-image.png",
+      timeToRead: "5 Min Read"
+    },
+  ]
+
+  getArticles() {
+    return this.articles.slice();
+  }
```
Comment (khanlifts) [inline src/app/services/articles.service.ts]: You could try this to make it reactive:
```suggestion
  getArticles$(start?: number): Observable<Article[]> {
    return of(this.articles.slice(start));
  }
```

File: src/app/articles/articles-list/article-pre-view/article-pre-view.component.ts (hunk 4 above)
Comment (khanlifts) [reply src/app/articles/articles-list/article-pre-view/article-pre-view.component.ts]: another empty line on line 10

File: src/app/pages/home/home.component.ts [hunk 11]
```diff
@@ -1,10 +1,11 @@
...
+import { Article } from "../../Interfaces/articleInterface";
 
 @Component({
   selector: 'app-home',
   templateUrl: './home.component.html',
   styleUrls: ['./home.component.scss']
 })
 export class HomeComponent {
-
+  articles: Article[];
```
Comment (khanlifts) [inline src/app/pages/home/home.component.ts]: Why did you decide to add this line?

File: src/styles/variables.scss [hunk 12]
```diff
@@ -12,6 +12,12 @@ $light-dark: rgba($dark, 0.88);
 $layout-spacing-small: 20px;
 $layout-spacing-medium: 40px;
 $layout-spacing-large: 95px;
+$border-radius-medium: 10px;
+$border-radius-small: 4px;
+$border-radius-map: (
+	small: $border-radius-small,
+	medium: $border-radius-medium,
+);
```
Comment (khanlifts) [inline src/styles/variables.scss]: Why do you think you need those separate (15/16) and in a map (17)?

File: src/app/pages/home/home.component.html [hunk 13]
```diff
@@ -16,6 +16,7 @@
       </div>
     </div>
 
-    <img class="intro__image" src="assets/icons/coding-illustration.svg">
+    <img class="intro__image" src="../../../assets/icons/coding-illustration.svg">
```
Comment (khanlifts) [inline src/app/pages/home/home.component.html]: Why did you add the `../../../` prefix?

File: src/app/pages/articles/articles-list/article-pre-view/article-pre-view.component.scss [hunk 14]
```diff
@@ -0,0 +1,57 @@
...
+		width: 250px;
+		@include border-radius(medium);
+	}
+
+	&__title {
+		@include headings(h2)
+	}
+
+	&:hover {
+		box-shadow: 0px 10px 15px 0px rgba(0, 0, 0, 0.2);
```
Comment (khanlifts) [inline src/app/pages/articles/articles-list/article-pre-view/article-pre-view.component.scss]: If you want to reuse the box-shadow you could also move it into a map.
This time you probably don't need a separate mixin and function to get it.

File: src/styles/variables.scss (hunk 12 above)
Comment (brunnermanu) [reply src/styles/variables.scss]: I needed the `$border-radius-small `variable in another map, but I remove the border-radius-map and only leave the variables. Since it doesn't make sense to have a map in this case.

File: src/app/pages/home/home.component.html (hunk 13 above)
Comment (brunnermanu) [reply src/app/pages/home/home.component.html]: I think my code editor did this... there is no reason for this change. I will undo it.
